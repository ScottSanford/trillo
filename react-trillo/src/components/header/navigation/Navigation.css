
/* USER NAVIGATION */
.user-nav {
    align-items: center;
    align-self: stretch; /* override the flex container align-items: center */
    display: flex;
    position: relative;

}

.user-nav > *:hover {
	background-color: var(--color-grey-light-2);
}

.user-nav__icon-box {
	align-items: center;
	display: flex;
	position: relative;
}

.user-nav__icon {
	fill: var(--color-grey-dark-2);
	height: 2.25rem;
	width: 2.25rem;
}

.user-nav__notification {
	background-color: var(--color-primary);
	border-radius: 50%;
	color: var(--color-white);
	font-size: .8rem;
	height: 1.75rem;
	position: absolute;
	top: 1.5rem;
	right: 1.1rem;
	width: 1.75rem;
	display: flex;
	align-items: center;
	justify-content: center;

}

@media only screen and (max-width: 31.25em) {
	.user-nav__notification {
		top: 2rem;
	}
}

.user-nav__user {
	align-items: center;
	display: flex;
}

.user-nav__user-photo {
	border-radius: 50%;
	height: 3.75rem;
	margin-right: 1rem;
}

.user-nav__notify:hover + .user-nav__notifications,
.user-nav__chat:hover + .user-nav__messages,
.user-nav__user:hover + .user-nav__dropdown {
	opacity: 1;
	visibility: visible;
	z-index: 100;
}

.user-nav--is-bold {
	font-weight: 600;
}

.user-nav__notifications,
.user-nav__messages,
.user-nav__dropdown {
	opacity: 0; /* TODO: need to fix this */
	padding: 0;
	position: absolute;
	top: 100%;
	visibility: hidden;
	transition: all .5s;
	z-index: 0;
}

.user-nav__dropdown {
	opacity: 0;
	visibility: visible;
	z-index: 100;
}

.user-nav__notifications {
	right: 16rem;
	width: 35rem;
}

.user-nav__messages {
	right: 10rem;
	width: 35rem;
}

.user-nav__dropdown {
	right: 0;
	width: 100%; /* parent element width */
}

.user-nav__dropdown:hover {
	opacity: 1;
	visibility: visible;
}

.user-nav__message-photo {
	border-radius: 50%;
	height: 5rem;
	margin-right: 1rem;
}

.user-nav__notification-link:link,
.user-nav__notification-link:visited,
.user-nav__message-link:link,
.user-nav__message-link:visited {
	align-items: center;
	border-bottom: var(--line-2);
	color: var(--color-grey-dark-1);
	display: flex;
	padding: 2rem;
	text-decoration: none;
	z-index: 10;
}

.user-nav__dropdown-link:link,
.user-nav__dropdown-link:visited {
	align-items: center;
	color: var(--color-grey-dark-1);
	display: flex;
	padding: 2.5rem 3.5rem;
	text-decoration: none;
	text-transform: uppercase;
	z-index: 10;
}

.user-nav__notification-link:link,
.user-nav__notification-link:visited {
	justify-content: space-between;
}

.user-nav__notification-message {
	flex: 0 0 69%;

	@media only screen and (min-width: $bp-largest) {
		flex: 0 0 65%;
	}
}

.user-nav__notification-date {
	align-self: flex-start;
}

.user-nav__see-all {
	align-items: center;
	background-color: var(--color-grey-light-1);
	display: flex;
	justify-content: center;
	padding: 1.5rem 2rem;
}

.user-nav__message-content {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
}

.user-nav__message-name {
	margin-right: auto;
	font-weight: 600;
}

.user-nav__message {
	color: var(--color-grey-dark-2);
	flex: 0 0 100%;
}

.user-nav__view-more {
	align-items: center;
	background-color: var(--color-grey-light-1);
	display: flex;
	justify-content: space-between;
	padding: 1.5rem 2rem;
}

.user-nav__notification-list,
.user-nav__message-list,
.user-nav__dropdown-list {
	list-style: none;
	position: relative;
}

.user-nav__notification-item,
.user-nav__message-item,
.user-nav__dropdown-item {
	background-color: var(--color-grey-light-2);
	cursor: pointer;
	transition: all .3s;
}

.user-nav__dropdown-item:hover {
	background-color: var(--color-grey-light-4);
}

.user-nav__dropdown-item:not(:last-child) {
	border-bottom: var(--line-2);
}

.user-nav__notification-icon,
.user-nav__dropdown-icon {
	fill: currentColor;
	height: 1.75rem;
	width: 1.75rem;
	margin-right: 2rem;
}

.user-nav > * {
	cursor: pointer;
	height: 100%;
	padding: 0 2rem;
}

@media only screen and (max-width: 31.25em) {
	.user-nav > * {
		padding: 2.1rem;
	}
}