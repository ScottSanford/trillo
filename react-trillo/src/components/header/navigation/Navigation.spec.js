import React from 'react'
import { shallow } from 'enzyme'
import Navigation from './Navigation'

describe('Navigation Component', () => {

	it('should match snapshot', () => {
		const wrapper = shallow(<Navigation />)
		expect(wrapper).toMatchSnapshot()
	})
})