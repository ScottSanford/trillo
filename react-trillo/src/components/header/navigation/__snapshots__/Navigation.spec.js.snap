// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should match snapshot 1`] = `
<nav
  className="user-nav"
>
  <div
    className="user-nav__icon-box user-nav__notify"
  >
    <svg
      className="user-nav__icon"
    >
      <use
        xlinkHref="img/sprite.svg#icon-bookmark"
      />
    </svg>
    <span
      className="user-nav__notification"
    >
      5
    </span>
  </div>
  <div
    className="user-nav__notifications"
  >
    <ul
      className="user-nav__notifications-list"
    >
      <li
        className="user-nav__notification-item"
      >
        <a
          className="user-nav__notification-link"
          href="#"
        >
          <svg
            className="user-nav__notification-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-aircraft-take-off"
            />
          </svg>
          <span
            className="user-nav__notification-message"
          >
            Check into your flight 
            <span
              className="user-nav--is-bold"
            >
              #1619
            </span>
            .
          </span>
          <span
            className="user-nav__notification-date"
          >
            Jun 20th
          </span>
        </a>
      </li>
      <li
        className="user-nav__notification-item"
      >
        <a
          className="user-nav__notification-link"
          href="#"
        >
          <svg
            className="user-nav__notification-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-map"
            />
          </svg>
          <span
            className="user-nav__notification-message"
          >
            Your 
            <span
              className="user-nav--is-bold"
            >
              AlgarExperience, Enjoy the Sea
            </span>
             reservation has been confirmed.
          </span>
          <span
            className="user-nav__notification-date"
          >
            Jun 18th
          </span>
        </a>
      </li>
      <li
        className="user-nav__notification-item"
      >
        <a
          className="user-nav__notification-link"
          href="#"
        >
          <svg
            className="user-nav__notification-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-key"
            />
          </svg>
          <span
            className="user-nav__notification-message"
          >
            Your 
            <span
              className="user-nav--is-bold"
            >
              Fiat 500
            </span>
             car reservation has been confirmed.
          </span>
          <span
            className="user-nav__notification-date"
          >
            Jun 18th
          </span>
        </a>
      </li>
      <li
        className="user-nav__notification-item"
      >
        <a
          className="user-nav__notification-link"
          href="#"
        >
          <svg
            className="user-nav__notification-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-home"
            />
          </svg>
          <span
            className="user-nav__notification-message"
          >
            Your hotel reservation at 
            <span
              className="user-nav--is-bold"
            >
              Hotel Las Palmas
            </span>
             have been confirmed.
          </span>
          <span
            className="user-nav__notification-date"
          >
            Jun 16th
          </span>
        </a>
      </li>
    </ul>
    <div
      className="user-nav__see-all"
    >
      <button
        className="btn-inline"
      >
        See all
      </button>
    </div>
  </div>
  <div
    className="user-nav__icon-box user-nav__chat"
  >
    <svg
      className="user-nav__icon"
    >
      <use
        xlinkHref="img/sprite.svg#icon-chat"
      />
    </svg>
    <span
      className="user-nav__notification"
    >
      13
    </span>
  </div>
  <div
    className="user-nav__messages"
  >
    <ul
      className="user-nav__message-list"
    >
      <li
        className="user-nav__message-item"
      >
        <a
          className="user-nav__message-link"
          href="#"
        >
          <img
            className="user-nav__message-photo"
            src="img/paige.jpeg"
          />
          <div
            className="user-nav__message-content"
          >
            <span
              className="user-nav__message-name"
            >
              Paige Spychalski
            </span>
            <span
              className="user-nav__message-date"
            >
              Jun 15th
            </span>
            <span
              className="user-nav__message"
            >
              Lorem ipsum dolor alem sit amet...
            </span>
          </div>
        </a>
      </li>
      <li
        className="user-nav__message-item"
      >
        <a
          className="user-nav__message-link"
          href="#"
        >
          <img
            className="user-nav__message-photo"
            src="img/user-1.jpg"
          />
          <div
            className="user-nav__message-content"
          >
            <span
              className="user-nav__message-name"
            >
              Nick Smith
            </span>
            <span
              className="user-nav__message-date"
            >
              Jun 8th
            </span>
            <span
              className="user-nav__message"
            >
              Lorem ipsum dolor alem sit amet...
            </span>
          </div>
        </a>
      </li>
      <li
        className="user-nav__message-item"
      >
        <a
          className="user-nav__message-link"
          href="#"
        >
          <img
            className="user-nav__message-photo"
            src="img/user-2.jpg"
          />
          <div
            className="user-nav__message-content"
          >
            <span
              className="user-nav__message-name"
            >
              Mary Thomas
            </span>
            <span
              className="user-nav__message-date"
            >
              May 21st
            </span>
            <span
              className="user-nav__message"
            >
              Lorem ipsum dolor alem sit amet...
            </span>
          </div>
        </a>
      </li>
      <li
        className="user-nav__message-item"
      >
        <a
          className="user-nav__message-link"
          href="#"
        >
          <img
            className="user-nav__message-photo"
            src="img/user-3.jpg"
          />
          <div
            className="user-nav__message-content"
          >
            <span
              className="user-nav__message-name"
            >
              Luke Walker
            </span>
            <span
              className="user-nav__message-date"
            >
              Jan 15th
            </span>
            <span
              className="user-nav__message"
            >
              Lorem ipsum dolor alem sit amet...
            </span>
          </div>
        </a>
      </li>
    </ul>
    <div
      className="user-nav__view-more"
    >
      <button
        className="btn-inline"
      >
        New message
      </button>
      <button
        className="btn-inline"
      >
        View all messages
      </button>
    </div>
  </div>
  <div
    className="user-nav__user"
  >
    <img
      alt="User photo"
      className="user-nav__user-photo"
      src="img/scott.jpeg"
    />
    <span
      className="user-nav__user-name"
    >
      Scott
    </span>
  </div>
  <div
    className="user-nav__dropdown"
  >
    <ul
      className="user-nav__dropdown-list"
    >
      <li
        className="user-nav__dropdown-item"
      >
        <a
          className="user-nav__dropdown-link"
          href="#"
        >
          <svg
            className="user-nav__dropdown-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-user"
            />
          </svg>
          <span>
            Profile
          </span>
        </a>
      </li>
      <li
        className="user-nav__dropdown-item"
      >
        <a
          className="user-nav__dropdown-link"
          href="#"
        >
          <svg
            className="user-nav__dropdown-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-suitcase"
            />
          </svg>
          <span>
            Reservations
          </span>
        </a>
      </li>
      <li
        className="user-nav__dropdown-item"
      >
        <a
          className="user-nav__dropdown-link"
          href="#"
        >
          <svg
            className="user-nav__dropdown-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-cog"
            />
          </svg>
          <span>
            Settings
          </span>
        </a>
      </li>
      <li
        className="user-nav__dropdown-item"
      >
        <a
          className="user-nav__dropdown-link"
          href="#"
        >
          <svg
            className="user-nav__dropdown-icon"
          >
            <use
              xlinkHref="img/sprite.svg#icon-log-out"
            />
          </svg>
          <span>
            Logout
          </span>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
